blazar/plugins/oshosts/host_plugin.py

```diff
@@ -973,6 +973,46 @@ def poll_resource_failures(self):
                 recovered_hosts.extend([host for host in unreservable_hosts
                                         if host['id'] in active_hv_ids])
 
+            aggregates = self.nova.aggregates.list()
+            # create a map to get the current aggregate of a host in nova
+            host_to_agg_map = {host: agg for agg in aggregates for host in agg.hosts}
+            # get all physical hosts
+            all_hosts = db_api.host_list()
+            pool = nova.ReservationPool()
+            freepool = pool.get_aggregate_from_name_or_id(pool.freepool_name)
+            if not freepool:
+                raise ValueError("Aggregate list does not contain a freepool!")
+            for host in all_hosts:
+                # get the most recent reservation for the host_id and check if we need to move to freepool
+                reservation = db_utils.get_recent_non_pending_reservation_by_host_id(host['id'])
+                if reservation and reservation["reservation_status"] == status.reservation.ACTIVE:
+                    LOG.info(f"{host['hypervisor_hostname']} is in a active reservation"
+                             f"{reservation['reservation_id']}")
+                    continue
+                # host not in 'active' or 'pending' reservation should be in freepool
+                host_uuid = host['hypervisor_hostname']
+                curr_agg = host_to_agg_map.get(host['hypervisor_hostname'])
+                if curr_agg and (curr_agg.name == freepool.name):
```

Comment:
Just convincing myself why we are only comparing to "freepool", rather than also to the target aggregate from the current reservationm.

`curr_agg` comes from `host_to_agg_map` which comes from `self.nova.aggregates.list()`
so, this is the value returned by nova?

this is in contrast to the value we get from the blazar DB, `models.ComputeHostReservation.aggregate_id`, which is the "target" aggregate, since that's only relevant during `active` or `pending` reservations, we ignore it, as we have skipped that case